python/graphstorm/model/gnn.py

```diff
@@ -140,12 +140,175 @@ class GSgnnModelBase(nn.Module):
+
+            for ntype, sparse_emb in sparse_embeds.items():
+                load_sparse_emb(sparse_emb, os.path.join(model_path, ntype),
+                                local_rank, world_size)
+
+        Parameters
+        ----------
+        restore_model_path : str
+            The path where we can restore the model.
+        local_rank: int
+            Local rank of the current process in a distributed env.
+        world_size : int
+            World size in a distributed env. This tells the size of a distributed cluster
+            (How many processes in a cluster).
+        """
+
+    @abc.abstractmethod
+    def save_dense_model(self, model_path):
+        """Save the dense models.
+
+        All model parameters except for learnable node embeddings, i.e.,
+        dgl.distributed.DistEmbedding, are saved by this function.
+        This fuction should go though all the sub NN models and save the correspoinding
+        parameters under ``model_path``.
+
+        Example:
+        --------
+
+        .. code::
+            # This function is only called by rank 0
+            input_encoder = self.input_encoder
+            gnn_model = self.gnn_model
+            decoder = self.decoder
+
+            model_states = {}
+            model_states['gnn'] = gnn_model.state_dict()
+            model_states['input'] = input_encoder.state_dict()
+            model_states['decoder'] = decoder.state_dict()
+
+            os.makedirs(model_path, exist_ok=True)
+            # mode 767 means rwx-rw-rwx:
+            os.chmod(model_path, 0o767)
+            th.save(model_states, os.path.join(model_path, 'model.bin'))
+
+        Parameters
+        ----------
+        model_path : str
+            The path where all model parameters and optimizer states are saved.
+        """
+
+    @abc.abstractmethod
+    def save_sparse_model(self, model_path, local_rank, world_size):
+        """Save sparse models, e.g., learnable node embeddings
+
+        Learnable node embeddings are saved by this function. Saving learnable
+        node embeddings only works when 1) the training task is run on a single machine
+        or 2) the training task is running on a distributed environment with a
+        shared file system.
+
+        Example:
+        --------
+        The implementation of save_sparse_model usually includes two steps:
+
+        Step 1: Create a path to save the learnable node embeddings. Please note,
+        only rank 0 does step 1.
+
+        .. code::
+            from graphstorm.model.util import create_sparse_emb_path
+
+            if local_rank == 0:
+                for ntype, sparse_emb in sparse_embeds.items():
+                    create_sparse_emb_path(model_path, ntype)
+            # make sure rank 0 creates the folder and change permission first
+            barrier()
+
+        Step 2: Save learnable node embeddings.
+        .. code::
+            from graphstorm.model.utils import save_sparse_embeds
```

Comment:
I think the imported function should be `save_sparse_emb`, because the line 281 use this function.